### PR TITLE
Add some tests and fix others

### PR DESCRIPTION
**Pull request by ljacobson64**
This PR adds 2 tests to test_TrackLengthMeshTally which make sure tetmeshes work for neutrons and protons. It also fixes some existing dagmcnp tests and adds them to the travis run script.

### CODE REVIEW

**Comment by ljacobson64** [comment]
Just commenting that the tests are expected to fail for this PR until #470 is merged, at which point the tests should pass

**Comment by makeclean** [comment]
#470 is now merged, can you rebase and push?

**Comment by ljacobson64** [comment]
Done, fingers crossed

**Comment by ljacobson64** [comment]
:(

**Comment by ljacobson64** [comment]
This is failing because the dagmcnp_unit_tests aren't being built, because we aren't using `-DBUILD_MCNP5` or `-DBUILD_MCNP6`, which we can't use because that would require us to build MCNP. The way around this would be to build the MCNP tests every time regardless of whether MCNP is being built.

**Comment by makeclean** [comment]
Yeah we should always build those in testing

**Comment by ljacobson64** [comment]
Perhaps a `-DBUILD_TESTS` flag which we would only use for the CI

**Comment by makeclean** [comment]
Good to go?

**Comment by ljacobson64** [comment]
Yes

File: mcnp/test/CMakeLists.txt [hunk 1]
```diff
@@ -5,17 +5,24 @@ set(LINK_LIBS)
 list(APPEND LINK_LIBS gtest)
 list(APPEND LINK_LIBS pthread)
 list(APPEND LINK_LIBS dagmciface)
+list(APPEND LINK_LIBS dagtally)
 list(APPEND LINK_LIBS ${DAG_LINK_LIBS})
 
 # Include directories
 include_directories(..)
+include_directories(../../common)
```
Comment (makeclean) [inline mcnp/test/CMakeLists.txt]: could you change this to ${CMAKE_SOURCE_DIR}/common

File: mcnp/test/CMakeLists.txt [hunk 2]
```diff
@@ -5,17 +5,24 @@ set(LINK_LIBS)
...
 include_directories(${MOAB_INCLUDE_DIRS})
 include_directories(${HDF5_INCLUDE_DIRS})
 include_directories(${GTEST_INCLUDE_DIR})
 
 # Setup unit tests
+set(TEST_NAME dagmcnp_unit_tests)
 set(DRIVER mcnp5_unit_test_driver.cc)
-setup_test(dagmcnp_unit_tests cpp ${DRIVER} "${LINK_LIBS}")
```
Comment (makeclean) [inline mcnp/test/CMakeLists.txt:18]: Are you changing this for all the tests? What was the reason for this change?

**Review by makeclean — CHANGES_REQUESTED** [review]

File: mcnp/test/CMakeLists.txt (hunk 2 above)
Comment (ljacobson64) [reply mcnp/test/CMakeLists.txt:18]: mcnp_funcs.cpp needs to be in there for this one, and the setup_test macro doesn't allow for adding additional source files

Comment (ljacobson64) [reply mcnp/test/CMakeLists.txt:18]: To elaborate, mcnp_funcs.cpp needs to be in there when you don't build DAG-MCNP for some reason

Comment (ljacobson64) [reply mcnp/test/CMakeLists.txt:18]:
```
CMakeFiles/dagmcnp_unit_tests.dir/dagmcnp_unit_tests.cpp.o: In function `DAGMCNP5Test_dagmcinit_test_Test::TestBody()':
dagmcnp_unit_tests.cpp:(.text+0x1cf): undefined reference to `dagmcwritemcnp_'
dagmcnp_unit_tests.cpp:(.text+0x3a6): undefined reference to `dagmc_teardown_'
```

File: mcnp/test/CMakeLists.txt (hunk 1 above)
Comment (ljacobson64) [reply mcnp/test/CMakeLists.txt]: Whoops, didn't see this one. Fixed now.

**Review by makeclean — COMMENTED** [review]
Awesome

**Review by makeclean — APPROVED** [review]

**Pull request merged**